api/controllers/ViewController.js

```diff
@@ -800,9 +800,92 @@ module.exports = {
     if (!cookie || (cookie && !cookie.userId) || (cookie && !cookie.totp)) {
       return reply.redirect('/');
     }
+
+    // Check for user feedback and display
+    let alert;
+    if (cookie.alert) {
+      alert = cookie.alert;
+      delete(cookie.alert);
+      request.yar.set('session', cookie);
+    }
+
+    // Load user from DB.
     const user = await User.findOne({ _id: cookie.userId });
+
+    // Render settins page.
     return reply.view('settings', {
       user,
+      alert,
     });
   },
+
+  /**
+   * Handle form submissions related to OAuth Client management.
+   */
+  async settingsOauthSubmit(request, reply) {
+    // If the user is not authenticated, redirect to the login page
+    const cookie = request.yar.get('session');
+    if (!cookie || (cookie && !cookie.userId) || (cookie && !cookie.totp)) {
+      return reply.redirect('/');
+    }
+
+    // Load current user from DB.
+    const user = await User.findOne({ _id: cookie.userId });
+
+    // Set up user feedback.
+    let reasons = [];
+    let alert = {};
+
+    console.log('🐛', request.payload);
```

Comment:
Is that supposed to be there?

Reply:
whoops, no never. the 🐛 is supposed to be my easy way to spot and remove debug stuff 🙃